### PR TITLE
Add Complex Type "Aggregate State"

### PR DESCRIPTION
**Pull request by koepalex**
## Purpose
* For demo scenario, we would like to have an easy example. The OPC PLC should deliver values for temperature and pressure and a "business application" can use those values to calculate the aggregate state of water (liquid, solid, fluid).
* The Simulation changes every 5 seconds randomly between the aggregation state

## Does this introduce a breaking change?
```
[ ] Yes
[ X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* clone the forked repo and build visual studio solution

* Test the code
```
* start opcplc with --ctas flag
* monitor node OPC UA node **Aggregate State #1**

```

## What to Check
Verify that the following are valid

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,6 +138,17 @@ Features:
...
+## Complex type (Aggregate State)
+
+The option `--ctas` adds a simple aggregate state to the address space.
+
+Features:
+- AggregateState is a complex type that shows: temperature and pressure
+- It changes every 5 seconds between different values for temperature and pressure leading to different aggregate states of water
+  - liquid
+  - solid
+  - fluid
```
Comment (luiscantero) [inline README.md:150]: Fluid is the same as liquid. Should be "gaseous".

File: src/PluginNodes/AggregateStatePluginNode.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
+﻿namespace OpcPlc.PluginNodes
```
Comment (luiscantero) [inline src/PluginNodes/AggregateStatePluginNode.cs:1]: Please use file-scoped namespace

File: src/PluginNodes/AggregateStatePluginNode.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+{
```
Comment (luiscantero) [inline src/PluginNodes/AggregateStatePluginNode.cs:2]: Should we rename to ComplexTypeAggregateStatePluginNode.cs for consistency?

File: src/PluginNodes/AggregateStatePluginNode.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            return predefinedNodes;
+        }
+
+        public void UpdateAggregateState(object state, ElapsedEventArgs elapsedEventArgs)
+        {
+            _pressureIndex++;
+
+            var newValue = new AggregateStateDataType
+            {
+                Temperature = 0,
```
Comment (luiscantero) [inline src/PluginNodes/AggregateStatePluginNode.cs:106]: The temperature never changes?

**Review by luiscantero — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (koepalex) [reply README.md:150]: thanks, updated

File: src/PluginNodes/AggregateStatePluginNode.cs (hunk 2 above)
Comment (koepalex) [reply src/PluginNodes/AggregateStatePluginNode.cs:1]: adopted to file-scoped namespace

File: src/PluginNodes/AggregateStatePluginNode.cs (hunk 3 above)
Comment (koepalex) [reply src/PluginNodes/AggregateStatePluginNode.cs:2]: reanmed

File: src/PluginNodes/AggregateStatePluginNode.cs (hunk 4 above)
Comment (koepalex) [reply src/PluginNodes/AggregateStatePluginNode.cs]: To change the aggregate state it is possible to define:

* temperature fix and changing only pressure 
* pressure fix and changing only temperatiure
* changing temperature and pressure

For simplicity I used fixed temperature. the other variants can be tested by using OPA UA Clients.

Comment (luiscantero) [reply src/PluginNodes/AggregateStatePluginNode.cs]: Ok, I got confused bcs the description in the readme says "It changes every 5 seconds between different values for temperature and pressure", we might need to clarify that better.

Comment (luiscantero) [reply src/PluginNodes/AggregateStatePluginNode.cs]: Also, pls update the readme - specifically the output of the help under command-line reference. You just need to add the new commands.

**Review by luiscantero — APPROVED** [review]

**Comment by hansgschossmann** [comment]
I like what you are trying to achieve here which is adding nodes to allow the business logic to use the node values to make advanced calculations.

I want to develop a consistent scenario, which can be even used for remote control in the future. This was one of the motivations to add a Boiler to opc-plc.

Can you pls extend the Boiler, which has a temperature already with a pressure and ideally already now with a variable node which can be written to change the function between temperature and pressure. Right now we only have methods to control, but we are not yet able to call those from our stack.

**Review by hansgschossmann — CHANGES_REQUESTED** [review]
:clock1:

**Pull request closed without merging**